test/x86/avx512f.c

```diff
@@ -1309,6 +1534,214 @@ test_simde_mm512_setone_si512(const MunitParameter params[], void* data) {
   return MUNIT_OK;
 }
 
+static MunitResult
+test_simde_mm512_srli_epi64(const MunitParameter params[], void* data) {
+  (void) params;
+  (void) data;
+
+  const struct {
+    simde__m512i a;
+    uint32_t     b;
```

Comment:
This type should match the parameter, which in this case is `unsigned int` not `uint32_t`.  `unsigned int` isn't 32-bits on all platforms; IIRC it's 16 bits on AVR (though I think that violates the C standard), and 64 bits on some platforms (ILP64 and SILP64).

Reply:
OK, I'll correct.

Reply:
Is it realistic to run AVX512 on a machine with 16-bit ints?
I haven't worked on a 16-bit machine for at least 30 years.
I agree that while K&R permitted almost any size of int, recent standards impose a minimum of 32 bits. In K&R IIRC the only restriction was sizeof(long) >= sizeof(int) >= sizeof(short) >= sizeof(char) plus a char must be at least 8 (or 7 for ASCII?) bits. (I have even worked on a machine with 32-bit chars.)
Anyway for the shifts (see below) I wanted to check the effect of setting the MSBit which would generate a negative value if cast to int.
But in doing this I assumed 32-bit ints :-(
Can I issue a PR with this potential problem?